cfy_manager/components/validations.py

```diff
@@ -458,38 +461,81 @@ def _validate_postgres_inputs():
+            config['postgresql_server']['key_path'] and
+            config['postgresql_server']['ca_path']):
+        if config[POSTGRESQL_SERVER][SSL_ENABLED] or \
+           config[POSTGRESQL_SERVER]['cluster']['nodes']:
             raise ValidationError(error_msg.format(
                 'a CA certificate, a certificate and a key',
-                'postgresql_server_cert_path, postgresql_server_key_path, '
-                'postgresql_ca_cert_path'))
+                'postgresql_server.cert_path, '
+                'postgresql_server.key_path, '
+                'and postgresql_server.ca_path'))
     elif not (config[SSL_INPUTS]['postgresql_client_cert_path'] and
               config[SSL_INPUTS]['postgresql_client_key_path'] and
-              config[SSL_INPUTS]['postgresql_ca_cert_path']):
+              config['postgresql_server']['ca_path']):
         if config[POSTGRESQL_CLIENT][SSL_CLIENT_VERIFICATION]:
             raise ValidationError(error_msg.format(
                 'with client verification, a CA certificate, '
                 'a certificate and a key ',
-                'postgresql_client_cert_path, postgresql_client_key_path, '
-                'postgresql_ca_cert_path'))
-    if not config[SSL_INPUTS]['postgresql_ca_cert_path'] and \
+                'ssl_inputs.postgresql_client_cert_path, '
+                'ssl_inputs.postgresql_client_key_path, '
+                'and postgresql_server.ca_path'))
+    if not config['postgresql_server']['ca_path'] and \
             config[POSTGRESQL_CLIENT][SSL_ENABLED]:
         raise ValidationError(error_msg.format('a CA certificate',
-                                               'postgresql_ca_cert_path'))
+                                               'postgresql_server.ca_path'))
 
 
-def _validate_external_postgres_ssl_enabled():
-    """For an external database, SSL for the DB must be enabled."""
+def _validate_external_postgres():
+    pg_conf = config[POSTGRESQL_SERVER]
+
     if _is_installed(DATABASE_SERVICE) and _is_installed(MANAGER_SERVICE):
-        return
-    if _is_installed(DATABASE_SERVICE) \
-            and not config[POSTGRESQL_SERVER][SSL_ENABLED]:
-        raise ValidationError('When installing an external database, SSL'
-                              ' must be enabled')
+        if pg_conf['cluster']['nodes']:
+            raise ValidationError('Postgres cluster nodes cannot be '
+                                  'installed on manager nodes.')
+        else:
+            # Local DB, no need to conduct external checks
+            return
+
+    if _is_installed(DATABASE_SERVICE):
+        if pg_conf['cluster']['nodes']:
+            problems = []
+
+            if len(pg_conf['cluster']['nodes']) != 3:
+                problems.append('There must be exactly three cluster nodes.')
+
+            etcd_conf = pg_conf['cluster']['etcd']
+            if not all(
+                etcd_conf[entry] for entry in
```

Comment:
`.get()` so that if it's not set at all, we don't get an ugly keyerror

Reply:
Fixed.